public/modules/custom/grants_profile/src/GrantsProfileService.php

```diff
@@ -644,4 +660,59 @@ private function decodeProfileContentRecursive(array $profileDocumentContent): a
     return $profileDocumentContent;
   }
 
+  /**
+   * The getUpdatedAt method.
+   *
+   * This method returns timestamp of the time
+   * a profile was last updated.
+   *
+   * @return string
+   *   Timestamp of last updated at.
+   */
+  public function getUpdatedAt() {
+    // Get selected company.
+    $selectedCompany = $this->getSelectedRoleData();
+    // Get grants profile.
+    $grantsProfileDocument = $this->getGrantsProfile($selectedCompany);
+
+    $profileUpdatedAt = $grantsProfileDocument->getUpdatedAt();
+    $profileUpdatedAt = strtotime($profileUpdatedAt);
+    return $profileUpdatedAt;
+  }
+
+  /**
+   * The getNotificationShown method.
+   *
+   * This method returns timestamp of the time
+   * a notification was shown.
+   *
+   * @return string
+   *   Timestamp of last time notification was shown.
+   */
+  public function getNotificationShown() {
+    // Get selected company.
+    $selectedCompany = $this->getSelectedRoleData();
+    // Get grants profile.
+    $grantsProfileDocument = $this->getGrantsProfile($selectedCompany);
+
+    $profileMetadata = $grantsProfileDocument->getMetadata();
+    $notification_shown = $profileMetadata['notification_shown'];
```

Comment:
```suggestion
    $notification_shown = $profileMetadata['notification_shown'] ?? 0;
```

Varmaan vois defaulttaa ton nollaks jos profiilista ei löydy, niin ei logita turhaan näitä viestejä 
![image](https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/43133397/cd87ec4b-bad1-46f3-a17f-044db1004165)